p2p/host/autonat/autonat_test.go

```diff
@@ -252,6 +239,21 @@ func TestAutoNATObservationRecording(t *testing.T) {
 		t.Fatalf("too-extreme private transition.")
 	}
 
+	// don't emit events on observed address change
+	newAddr, _ := ma.NewMultiaddr("/ip4/127.0.0.1/udp/12345")
+	an.recordObservation(autoNATResult{network.ReachabilityPublic, newAddr})
+	if an.Status() != network.ReachabilityPublic {
+		t.Fatalf("reachability should stay public")
+	}
+	select {
+	case e := <-s.Out():
+		_, ok := e.(event.EvtLocalReachabilityChanged)
```

Comment:
Nit: remove the `ok` assertion here. It's ok to panic (especially since it's a test), and would make the test more robust.

Reply:
So I removed the assertion completely. The subscription is for reachability change, any event received here is an error in this case.